app/components/password/reset-password.js

```diff
@@ -0,0 +1,46 @@
+import Ember from 'ember';
+const { Component, set, get } = Ember;
+import { task } from 'ember-concurrency';
+
+export default Component.extend({
+  /**
+   * @property password
+   * @default String
+   */
+  password: '',
+  /**
+   * @property passwordConfirmation
+   * @default String
+   */
+  passwordConfirmation: '',
+  /**
+   * @property error
+   */
+  error: null,
+
+  /**
+   * @property resetPasswordTask
+   * @param password
+   * @param passwordConfirmation
+   */
+  resetPasswordTask: task(function* (password, passwordConfirmation) {
+    try {
+      yield get(this, 'resetPassword')(password, passwordConfirmation);
```

Comment:
Also, this is where I would clear the error. Simply do a `set(this, 'error', null)` before the `yield`.